packages/expo-modules-core/common/cpp/fabric/ExpoViewProps.cpp

```diff
@@ -17,10 +17,11 @@ std::unordered_map<std::string, folly::dynamic> propsMapFromProps(const ExpoView
 
   // Iterate over values in the raw props object.
   // Note that it contains only updated props.
-  rawProps.iterateOverValues([&propsMap](react::RawPropsPropNameHash hash, const char *name, const react::RawValue &value) {
-    std::string propName(name);
-    propsMap[propName] = (folly::dynamic)value;
-  });
+  const auto& dynamicRawProps = static_cast<folly::dynamic>(rawProps);
+  for (const auto& propsPair : dynamicRawProps.items()) {
+    std::string propName = propsPair.first.getString();
+    propsMap[propName] = (folly::dynamic)propsPair.second;
```

Comment:
```suggestion
    propsMap[propName] = static_cast<folly::dynamic>(propsPair.second);
```
Let's use the static cast instead of a `C` like cast